inc/admin.php

```diff
@@ -2265,4 +2278,28 @@ public function should_show_exceed_quota_warning() {
 
 		return true;
 	}
+
+	/**
+	 * Mark the users that are on free plan and have offload enabled.
+	 *
+	 * Used for displaying a notice after plugin update about offload restrictions.
+	 *
+	 * @return void
+	 */
+	public function mark_free_user_with_offload() {
+		if ( ! $this->settings->is_connected() ) {
+			return;
+		}
+
+		$service_data = $this->settings->get( 'service_data' );
+		if ( isset( $service_data['plan'] ) && 'free' !== $service_data['plan'] ) {
```

Comment:
here I dont see using the hosting status to mark this.

Reply:
By 'hosting status,' do you mean the `'status'` property?